examples/build_all.py

```diff
@@ -1,29 +1,46 @@
 #!/usr/bin/python3
 
+import os
+
 from pathlib import Path
 from subprocess import PIPE, run
 
+# NOTE: boost V1.67 is to old! CK
 examples = [
-  x for x in Path(__file__).parent.iterdir() if x.is_dir() and (x / 'CMakeLists.txt').exists()
+    x for x in Path(__file__).parent.iterdir() if x.is_dir() and (x / 'CMakeLists.txt').exists() and (not x.name in ['boost', 'old-gtest'])
 ]
 
 assert(len(examples) > 0)
 
+
 def runCommand(command):
-  print('- %s' % command)
-  result = run(command, stdout=PIPE, stderr=PIPE, universal_newlines=True, shell=True)
-  if result.returncode != 0:
-    print("error while running '%s':\n" % command, '  ' + str(result.stderr).replace('\n','\n  '))
-    exit(result.returncode)
-  return result.stdout
+    print('- %s' % command)
+    result = run(command, stdout=PIPE, stderr=PIPE, universal_newlines=True, shell=True)
+    if result.returncode != 0:
+        print("error while running '%s':\n" %
+              command, '  ' + str(result.stderr).replace('\n', '\n  '))
+        exit(result.returncode)
+    return result.stdout
+
 
 print('')
 for example in examples:
-  print("running example %s" % example.name)
-  print("================" + ('=' * len(example.name)))
-  project = Path(".") / 'build' / example.name
-  configure = runCommand('cmake -H%s -B%s' % (example, project))
-  print('  ' + '\n  '.join([line for line in configure.split('\n') if 'CPM:' in line]))
-  build = runCommand('cmake --build %s -j4' % (project))
-  print('  ' + '\n  '.join([line for line in build.split('\n') if 'Built target' in line]))
-  print('')
+    print("running example %s" % example.name)
+    print("================" + ('=' * len(example.name)))
+    project = Path(".") / 'build' / example.name
+    #
+    # Note: needs at least cmake V3.15! CK
+    # https://cmake.org/cmake/help/latest/command/project.html#code-injection
+    #
+    cmakeModulesPath = os.environ['HOME'] + '/Workspace/cmake'
+    if Path(cmakeModulesPath).is_dir():
+        before = "-DCMAKE_PROJECT_INCLUDE_BEFORE=%s/before_project_setup.cmake" % (
+            cmakeModulesPath)
+        after = "-DCMAKE_PROJECT_INCLUDE=%s/build_options.cmake" % (cmakeModulesPath)
+        configure = runCommand('cmake -H%s -B%s -G Ninja %s %s' % (example, project, before, after))
+    else:
+        configure = runCommand('cmake -H%s -B%s' % (example, project))
```

Comment:
This seems to be oddly workspace-specific. Is it really necessary?

Reply:
yes, some of the creates features of cmake:

I use it to setup the ccache, cxx std , warnings , clang-tidy, inject my modules want to have, ...

Reply:
I do see the value of this feature, however I'm not sure how it fits into the automated example script here, as there are also many other combinations of arguments that users could potentially pass to CMake that we can't possible all handle explicitly. As a compromise, how about forwarding any extra args to the python script to CMake? Then you could invoke the script using

```bash
./build_all.py -DCMAKE_PROJECT_INCLUDE_BEFORE=my/before/script -DCMAKE_PROJECT_INCLUDE=my/after/script
```

Reply:
good idea ...

I will try it